### ISSUE TITLE
Can't format Gradle files

### ISSUE THREAD

**Issue by NatoBoram** [badsyntax/vscode-spotless-gradle#185]
I'm trying to format Gradle files, but I have this error.

> There is no formatter for 'gradle' files installed.

![image](https://user-images.githubusercontent.com/10495562/97380974-7e6e7680-189e-11eb-8573-765831ae2bfc.png)

How do we use this extension?

**Comment by badsyntax** [comment]
The extension currently won't format gradle files. 

Just to confirm, have you setup spotless to format gradle files? Can spotless actually format gradle files?

**Comment by NatoBoram** [comment]
The issue is that it's marked as a formatter in the VSCode marketplace but it doesn't actually format anything

**Comment by badsyntax** [comment]
That statement is obviously not true.

It will format these file types: https://github.com/badsyntax/vscode-spotless-gradle/blob/4c3d516a60b7b2e114d95b3377cfccc90834ca51/src/constants.ts#L1-L26

..but the formatting entirely depends on how you've setup spotless. I get the feeling you don't really understand the point of this extension. Is there anything I can do to make it clearer?

**Comment by badsyntax** [comment]
I see the confusion now. VS Code marketplace is showing that this extension should format gradle files. It looks like it's pulling that from the keywords.: https://github.com/badsyntax/vscode-spotless-gradle/blob/4c3d516a60b7b2e114d95b3377cfccc90834ca51/package.json#L40

I'm not sure how I can configure this so that it doesn't show this as a formatter for gradle file types. I will do some investigation though.

**Comment by badsyntax** [comment]
It looks like this is a limitation of the marketplace. I might have to remove the "gradle" keyword but I'd prefer not to.

**Comment by NatoBoram** [comment]
Huh, I'm surprised that the marketplace tries to infer unrelated tags using keywords. Sounds like a recipe for disappointment.

**Comment by badsyntax** [comment]
Yea, it's a pity there's not much granularity here. 

I've reached out to the vscode community on slack about this, and confirmed there's no way to define supported languages for formatters as part of the extension metadata. I'm going to try speak with the marketplace authors to see if they're wiling to improve this (but imo unlikely they'll change this, but worth letting them know). 

Thanks for this bug report btw, I wasn't aware of this, and apologies for not initially understanding the bug report.

**Comment by NatoBoram** [comment]
Thanks! If you raise some issues, you can link them here and I'll follow their progress.

**Comment by badsyntax** [comment]
This should be fixed with [1.2.0](https://github.com/badsyntax/vscode-spotless-gradle/releases/tag/1.2.0). I'm sorry it took this long.

To enable Gradle file support in spotless:

```groovy
spotless {
  groovyGradle {
    target '*.gradle' // default target of groovyGradle
    greclipse()
  }
}
```

(See https://github.com/diffplug/spotless/tree/main/plugin-gradle#groovy)

To enable in vs code, add to `settings.json`:

```json
"[gradle]": {
  "editor.defaultFormatter": "richardwillis.vscode-spotless-gradle",
  "spotlessGradle.diagnostics.enable": true,
  "spotlessGradle.format.enable": true
}
```

It should be working now but let me know if you have any issues.
